app/components/EditProfileForm.tsx

```diff
@@ -1,17 +1,54 @@
 'use client'
 import { FormEvent } from 'react';
 
-export default function EditProfileForm() {
-
-  /**
-   * Submits new post data
-   * 
-   * @param e - the form submit event
-  */
-  const onSubmitHandler = (e: FormEvent<HTMLFormElement>): void => {
+export default function EditProfileForm({userData}) {
+
+  const firstName = userData['body']['firstName'][0]
+  const lastName = userData['body']['lastName'][0]
+  const username = userData['body']['username'][0]
+  const gender = userData['body']['gender'][0]
+  const bio = userData['body']['bio'][0]
+
+  const userInfo = {
+    'firstName' : firstName,
+    'lastName' : lastName,
+    'username' : username,
+    'gender' : gender,
+    'bio' : bio
+  };
+
+  const onSubmitHandler = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const formData: Object = Object.fromEntries((new FormData(e.currentTarget)));
-    console.log(JSON.stringify(formData));
+    
+    let attributes = {};
+
+    for (const [key, value] of Object.entries(formData)) {
+      // TODO: Add support for profile picture
+      if (key === "profilePicture") {
+        continue;
+      }
+      // If the value is the same, don't add to the request
+      if (userInfo[key] === value) {
+        continue;
+      } else {
+        attributes[key] = value;
+      }
+    }
+
+    const payload = {
+      "method" : "PUT",
+      "attributes" : attributes
+    }
+
+    const res = await fetch(`http://localhost:3000/api/users/${username}`, {
```

Comment:
Can you redirect user router.push() as seen in this [article?](https://nextjs.org/docs/app/api-reference/functions/use-router)

Reply:
Tried, didn't work. I was getting a "router not mounted error". I will probably have to make a separate function and pass it in to get it to work.